### PR TITLE
[Code health] Consolidate and rename test data

### LINKED ISSUE #1041: [Code health] Duplicate test data in several tests
`FeatureRepositoryTest` can use `FakeData` instead of redeclaring fake data for tests.

@shobhitagarwal1612 It looks like you added `FeatureRepositoryTest` before `FakeData` was introduced. Could you kindly refactor when you have a cycle or two? 🙏  Thank you!

### PR DESCRIPTION
**Pull request by gino-m**
Refactor, merge duplicates, and rename fake data constants.

Also simplifies setting of test project in `FakeRemoteDataStore`, removing conditional logic (https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html). 

Fixes #1041.

@shobhitagarwal1612 PTAL?

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
The instrumentation tests are failing. Could you please fix that?

```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NullPointerException: item is null
       FATAL EXCEPTION: main
Process: com.google.android.gnd, PID: 7234
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NullPointerException: item is null
	at io.reactivex.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
	at io.reactivex.internal.operators.mixed.FlowableSwitchMapCompletable$SwitchMapCompletableObserver.onError(FlowableSwitchMapCompletable.java:133)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.checkTerminated(FlowableObserveOn.java:209)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnSubscriber.runAsync(FlowableObserveOn.java:399)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
	at com.squareup.rx2.idler.DelegatingIdlingResourceScheduler$ScheduledWork.run(DelegatingIdlingResourceScheduler.java:142)
	at io.reactivex.internal.schedulers.TrampolineScheduler$TrampolineWorker.enqueue(TrampolineScheduler.java:112)
	at io.reactivex.internal.schedulers.TrampolineScheduler$TrampolineWorker.schedule(TrampolineScheduler.java:81)
	at com.squareup.rx2.idler.DelegatingIdlingResourceScheduler$1.schedule(DelegatingIdlingResourceScheduler.java:48)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.trySchedule(FlowableObserveOn.java:166)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.onError(FlowableObserveOn.java:128)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.onError(BasicFuseableSubscriber.java:101)
	at io.reactivex.processors.BehaviorProcessor$BehaviorSubscription.test(BehaviorProcessor.java:634)
	at io.reactivex.processors.BehaviorProcessor$BehaviorSubscription.emitNext(BehaviorProcessor.java:620)
	at io.reactivex.processors.BehaviorProcessor.onError(BehaviorProcessor.java:294)
	at io.reactivex.internal.operators.flowable.FlowableOnBackpressureLatest$BackpressureLatestSubscriber.checkTerminated(FlowableOnBackpressureLatest.java:159)
	at io.reactivex.internal.operators.flowable.FlowableOnBackpressureLatest$BackpressureLatestSubscriber.drain(FlowableOnBackpressureLatest.java:121)
	at io.reactivex.internal.operators.flowable.FlowableOnBackpressureLatest$BackpressureLatestSubscriber.onError(FlowableOnBackpressureLatest.java:75)
	at io.reactivex.internal.operators.flowable.FlowableSwitchMap$SwitchMapSubscriber.drain(FlowableSwitchMap.java:220)
	at io.reactivex.internal.operators.flowable.FlowableSwitchMap$SwitchMapSubscriber.onError(FlowableSwitchMap.java:142)
	at io.reactivex.internal.operators.flowable.FlowableSwitchMap$SwitchMapSubscriber.onNext(FlowableSwitchMap.java:117)
	at io.reactivex.internal.operators.flowable.FlowableDistinctUntilChanged$DistinctUntilChangedSubscriber.tryOnNext(FlowableDistinctUntilChanged.java:101)
	at io.reactivex.internal.operators.flowable.FlowableDistinctUntilChanged$DistinctUntilChangedSubscriber.onNext(FlowableDistinctUntilChanged.java:67)
	at io.reactivex.processors.PublishProcessor$PublishSubscription.onNext(PublishProcessor.java:360)
	at io.reactivex.processors.PublishProcessor.onNext(PublishProcessor.java:243)
	at com.google.android.gnd.repository.ProjectRepository.activateProject(ProjectRepository.java:144)
	at com.google.android.gnd.ui.home.-$$Lambda$CUuMpA9BeAhLeQ8uMhPtKAZzY3U.accept(Unknown Source:4)
	at java8.util.Optional.ifPresent(Optional.java:176)
	at com.google.android.gnd.ui.home.HomeScreenViewModel.init(HomeScreenViewModel.java:301)
	at com.google.android.gnd.ui.home.HomeScreenFragment.onStart(HomeScreenFragment.java:421)
	at androidx.fragment.app.Fragment.performStart(Fragment.java:3021)
	at androidx.fragment.app.FragmentStateManager.start(FragmentStateManager.java:589)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:300)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2106)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at androidx.test.espresso.base.Interrogator.loopAndInterrogate(Interrogator.java:53)
	at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:155)
	at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:149)
	at androidx.test.espresso.base.UiControllerImpl.injectMotionEvent(UiControllerImpl.java:53)
	at androidx.test.espresso.action.MotionEvents.sendUp(MotionEvents.java:122)
	at androidx.test.espresso.action.MotionEvents.sendUp(MotionEvents.java:117)
	at androidx.test.espresso.action.Tap.sendSingleTap(Tap.java:27)
	at androidx.test.espresso.action.Tap.access$100(Tap.java:21)
	at androidx.test.espresso.action.Tap$1.sendTap(Tap.java:3)
	at androidx.test.espresso.action.GeneralClickAction.perform(GeneralClickAction.java:23)
	at androidx.test.espresso.ViewInteraction$SingleExecutionViewAction.perform(ViewInteraction.java:16)
	at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:65)
	at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:15)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:3)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:2)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.NullPointerException: item is null
	at io.reactivex.internal.functions.ObjectHelper.requireNonNull(ObjectHelper.java:39)
	at io.reactivex.Single.just(Single.java:839)
	at com.google.android.gnd.persistence.remote.FakeRemoteDataStore.loadProject(FakeRemoteDataStore.java:64)
	at com.google.android.gnd.repository.ProjectRepository.syncProjectWithRemote(ProjectRepository.java:117)
	at com.google.android.gnd.repository.ProjectRepository.activateProject(ProjectRepository.java:99)
	at com.google.android.gnd.repository.ProjectRepository.lambda$ZaNpol0hpmmz0epk0c_iLE5Zgtc(Unknown Source:0)
	at com.google.android.gnd.repository.-$$Lambda$ProjectRepository$ZaNpol0hpmmz0epk0c_iLE5Zgtc.apply(Unknown Source:4)
	at io.reactivex.internal.operators.flowable.FlowableSwitchMap$SwitchMapSubscriber.onNext(FlowableSwitchMap.java:113)

```

**Review by shobhitagarwal1612 — DISMISSED** [review]

File: gnd/src/androidTest/java/com/google/android/gnd/AcceptTermsOfServiceTest.java [hunk 1]
```diff
@@ -33,11 +34,13 @@
...
 
   @Inject FakeAuthenticationManager fakeAuthenticationManager;
+  @Inject FakeRemoteDataStore fakeRemoteDataStore;
 
   @Override
   public void setUp() {
     super.setUp();
-    fakeAuthenticationManager.setUser(FakeData.TEST_USER);
+    fakeAuthenticationManager.setUser(FakeData.USER);
+    fakeRemoteDataStore.setTestProject(FakeData.PROJECT);
```
Comment (gino-m) [inline gnd/src/androidTest/java/com/google/android/gnd/AcceptTermsOfServiceTest.java:43]: @shobhitagarwal1612  It appears that despite this call, the testProject in `FakeRemoteDataStore` is still `null`. Any idea what I could be doing wrong?

Comment (shobhitagarwal1612) [reply gnd/src/androidTest/java/com/google/android/gnd/AcceptTermsOfServiceTest.java:43]: If the project is used in the constructor of hilt injected classes, then the test project should be set before the setup() call. This points to another problem in the project where some business logic is present inside the constructors (e.g. feature sync and signin observser in MainViewModel)

Comment (gino-m) [reply gnd/src/androidTest/java/com/google/android/gnd/AcceptTermsOfServiceTest.java:43]: I tried to work around that but couldn't find an easy way, so I field #1045 and set default values in the fakes themselves for now.

**Pull request merged**